pkg/sql/parser/sql.y

```diff
@@ -4163,6 +4163,14 @@ create_tenant_stmt:
   {
     $$.val = &tree.CreateTenant{Name: tree.Name($3)}
   }
+| CREATE TENANT name FROM REPLICATION OF name ON string_or_placeholder
```

Comment:
I think this statement might be the one where we really end up wanting to use placeholders for these names.  But, no need to change it now, we can do a follow-up PR where we just change them.